src/main/java/models/League.java

```diff
@@ -0,0 +1,46 @@
+package models;
+
+import java.util.LinkedList;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Queue;
+
+public class League {
+
+    private Queue<Team> initialTeams;
+    private HashMap<Integer, ArrayList<Team>> winners;
+
+    public League(Queue<Team> initialTeams){
+        this.initialTeams = initialTeams;
+        this.winners = new HashMap<>();
+    }
+
+    private void playRound(Queue<Team> teamsToPlay, int round){
+
+        Queue<Team> nextRoundTeams = new LinkedList<>();
+
+        if( winners.get(round) == null ){
+            winners.put(round, new ArrayList<>());
+        }
+
+        while( !teamsToPlay.isEmpty() ){
+            Match match = new Match(teamsToPlay.poll(), teamsToPlay.poll());
+            Team winner = match.getMatchWinner();
```

Comment:
Take care no issues arise later if a player does not have a initial score! `match.getMatchWinner()` -> `team.getTeamScore()` -> `player.getScore()`. `Player` does not have a default score of (for instance) 0!